modules/tinymce/src/plugins/lists/test/ts/browser/ApplyListOnCustomList.ts

```diff
@@ -0,0 +1,42 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Strings } from '@ephox/katamari';
+import { TinyAssertions, TinyHooks, TinyUiActions } from '@ephox/mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/lists/Plugin';
+
+describe('browser.tinymce.plugins.lists.ApplyListOnParagraphWithStylesTest', () => {
+
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    indent: false,
+    plugins: 'lists',
+    toolbar: 'numlist bullist',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ]);
+
+  it('TINY-9998: Apply list to custom list should keep the original list structure', () => {
```

Comment:
This test is quite dense could it be separated into the different cases being tested here?

Reply:
I split this test into two cases, nested lists and flat lists. How do you like the tests now ? Should they split even more or is this okay ? @spocke